.eslintrc.js

```diff
@@ -400,10 +400,185 @@ module.exports = {
+          {
+            // prevents UI code from importing server side code and then webpack including it when doing builds
+            patterns: ['**/server/*'],
+          },
+        ],
+      },
+    },
+    {
+      // typescript only for front and back end
+      files: ['x-pack/plugins/siem/**/*.{ts,tsx}'],
+      rules: {
+        // This will be turned on after bug fixes are complete
+        // '@typescript-eslint/explicit-member-accessibility': 'warn',
+        '@typescript-eslint/no-this-alias': 'error',
         '@typescript-eslint/no-explicit-any': 'error',
-        'import/order': 'error',
+        '@typescript-eslint/no-useless-constructor': 'error',
+        // This will be turned on after bug fixes are complete
+        // '@typescript-eslint/no-object-literal-type-assertion': 'warn',
+        '@typescript-eslint/unified-signatures': 'error',
+
+        // eventually we want this to be a warn and then an error since this is a recommended linter rule
+        // for now, keeping it commented out to avoid too much IDE noise until the other linter issues
+        // are fixed in the next release or two
+        // '@typescript-eslint/explicit-function-return-type': 'warn',
+
+        // these rules cannot be turned on and tested at the moment until this issue is resolved:
+        // https://github.com/prettier/prettier-eslint/issues/201
+        // '@typescript-eslint/await-thenable': 'error',
+        // '@typescript-eslint/no-non-null-assertion': 'error'
+        // '@typescript-eslint/no-unnecessary-type-assertion': 'error',
+        // '@typescript-eslint/no-unused-vars': 'error',
+        // '@typescript-eslint/prefer-includes': 'error',
+        // '@typescript-eslint/prefer-string-starts-ends-with': 'error',
+        // '@typescript-eslint/promise-function-async': 'error',
+        // '@typescript-eslint/prefer-regexp-exec': 'error',
+        // '@typescript-eslint/promise-function-async': 'error',
+        // '@typescript-eslint/require-array-sort-compare': 'error',
+        // '@typescript-eslint/restrict-plus-operands': 'error',
+        // '@typescript-eslint/unbound-method': 'error',
+      },
+    },
+    {
+      // typescript and javascript for front and back end
+      files: ['x-pack/plugins/siem/**/*.{js,ts,tsx}'],
+      plugins: ['react'],
+      rules: {
+        'accessor-pairs': 'error',
+        'array-callback-return': 'error',
+        'no-array-constructor': 'error',
+        // This will be turned on after bug fixes are mostly completed
+        // 'arrow-body-style': ['warn', 'as-needed'],
+        complexity: 'warn',
+        // This will be turned on after bug fixes are mostly completed
+        // 'consistent-return': 'warn',
+        // This will be turned on after bug fixes are mostly completed
+        // 'func-style': ['warn', 'expression'],
+        // These will be turned on after bug fixes are mostly completed and we can
+        // run a fix-lint
+        /*
+        'import/order': [
```

Comment:
Excited to get this back in as well. This is going to be a big change though, so I think holding off for now is 👍